### DISCUSSION TITLE
CKAN 2.10.3 CSRF token missing when upload in datastore

### DISCUSSION THREAD

**Discussion by ziorick** [ckan/ckan#8127] [Q&A]
Hi to all.
When I try to upload to datastore a resource via UI as in pic:
![immagine](https://github.com/ckan/ckan/assets/15434275/497ee25a-b9a8-4afe-a849-da22cefaefac)

I obtain the message (in browser and in ckan log):
![immagine](https://github.com/ckan/ckan/assets/15434275/7436855d-c441-46ca-9526-af4746f88fda)

I built my docker image from source and run via docker compose. All works well (harvesting too).

My pip list output is:
Package               Version     Editable project location
--------------------- ----------- -------------------------------------------
alembic               1.13.1
attrs                 23.2.0
Babel                 2.14.0
Beaker                1.12.1
bleach                5.0.1
blinker               1.7.0
boto3                 1.34.56
botocore              1.34.56
certifi               2024.2.2
chardet               5.2.0
charset-normalizer    2.0.12
ckan                  2.10.3      /usr/lib/ckan/default/src/ckan
ckanext-dcat          1.6.0       /usr/lib/ckan/default/src/ckanext-dcat
ckanext-geoview       0.1.0
ckanext-harvest       1.5.6       /usr/lib/ckan/default/src/ckanext-harvest
ckanext-hierarchy     1.2.1a0     /usr/lib/ckan/default/src/ckanext-hierarchy
ckanext-pdfview       0.0.6
ckanext-report        0.1         /usr/lib/ckan/default/src/ckanext-report
ckanext-xloader       0.12.2
ckantoolkit           0.0.7
click                 8.1.7
Deprecated            1.2.14
dominate              2.9.1
et-xmlfile            1.1.0
feedgen               0.9.0
Flask                 2.0.3
Flask-Babel           1.0.0
Flask-Login           0.6.3
flask-multistatic     1.0
Flask-WTF             1.0.1
future                1.0.0
geomet                1.1.0
greenlet              2.0.2
idna                  3.6
ijson                 3.2.3
importlib-metadata    4.11.3
isodate               0.6.1
itsdangerous          2.1.2
Jinja2                3.1.3
jmespath              1.0.1
jsonlines             4.0.0
linear-tsv            1.1.0
lxml                  4.9.1
Mako                  1.3.2
Markdown              3.5.2
MarkupSafe            2.1.5
mypy                  0.971
mypy-extensions       0.4.3
nose                  1.3.7
openpyxl              3.1.2
packaging             23.2
passlib               1.7.4
pika                  1.3.2
pip                   24.0
polib                 1.2.0
psycopg2              2.9.9
PyJWT                 2.8.0
pyparsing             3.1.2
pysolr                3.9.0
python-dateutil       2.9.0.post0
python-magic          0.4.27
pytz                  2024.1
pytz-deprecation-shim 0.1.0.post0
PyUtilib              6.0.0
PyYAML                6.0.1
rdflib                6.1.1
redis                 4.1.4
requests              2.31.0
rq                    1.16.0
s3transfer            0.10.0
setuptools            69.1.1
simplejson            3.19.2
six                   1.16.0
SQLAlchemy            1.4.41
sqlalchemy2-stubs     0.0.2a27
sqlparse              0.4.4
tabulator             1.53.5
tomli                 2.0.1
typing_extensions     4.10.0
tzdata                2024.1
tzlocal               5.2
unicodecsv            0.14.1
Unidecode             1.0.22
urllib3               1.26.9
watchdog              2.1.6
webassets             2.0
webencodings          0.5.1
Werkzeug              2.0.3
wrapt                 1.16.0
WTForms               3.1.2
xlrd                  2.0.1
zipp                  3.17.0
zope.interface        5.4.0

Can You help me?

**Comment by amercader** [comment]
I can't reproduce this locally @ziorick. Is there an error in the logs that can help us debug this?
